### PR TITLE
ICU-22220 Add step for updating API Doc table

### PR DESCRIPTION
**Pull request by sven-oly**
This adds steps to revise the table of API docs in README.md with updated version numbers for either the Dev or Released entries.

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22220
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by markusicu — CHANGES_REQUESTED** [review]
You are targeting the main branch with the 73rc BRS ticket. main goes into 74 now.
Please either retarget your PR to the maint/maint-73 branch, or use a ticket like ICU-22323 or ICU-22325.

**Comment by sven-oly** [comment]
I think I've addressed the comments. Ready to go?

File: docs/processes/release/tasks/docs.md [hunk 1]
```diff
@@ -396,6 +399,11 @@ See [Upload API documentations](docs.md#upload-api-documentations) below for how
 See <https://unicode-org.github.io/icu-docs/HOWTO-Update.html> for instructions
 to upload to <https://unicode-org.github.io/icu-docs/>
 
+#### Update table os API docs:
```
Comment (markusicu) [inline docs/processes/release/tasks/docs.md:402]:
```suggestion
#### Update table of API docs:
```

**Comment by sven-oly** [comment]
I think I've adopted the suggestions. PTAL.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**